### DISCUSSION TITLE
database connections are dropped and recreated during istiod restart or image updated

### DISCUSSION THREAD

**Discussion by parshva-shah-dsw** [istio/istio#53858] [General]
we are observing existing db connections being dropped and recreated at the application side when istiod pod is deleted or image is updated. 
<img width="677" alt="image" src="https://github.com/user-attachments/assets/08f0fd5f-21c4-436c-91e5-44537b1b5b42">

we need to understand from Istio perspective why existing db connections are refreshed is it an issue observed by anyone else also?

Istio :- 1.23.0
EKS :- 1.29.5

**Comment by parshva-shah-dsw** [comment]
@howardjohn @ilrudie  any idea on this issue ?

**Comment by ilrudie** [comment]
What is the data path? Is this all internal to a single cluster or traversing an ingress? Are you using sidecars or ambient?

**Reply by parshva-shah-dsw** [reply]
for Database we have service entry to connect the RDS. Rest are applications are using sidecar proxy for service mesh. 
We have single cluster using EKS.

**Reply by ilrudie** [reply]
Are you using [DNS proxy](https://istio.io/latest/docs/ops/configuration/traffic-management/dns-proxy/)?

Would you be willing to post the contents of your ServiceEntry (you can redact sensitive hostnames/addresses/etc)? If not publicly would you be able to DM me in the istio slack?

**Reply by parshva-shah-dsw** [reply]
we are not using DNS proxying at istio side.

Service entry contents 

apiVersion: networking.istio.io/v1
kind: ServiceEntry
metadata:
  name: external-rds
spec:
exportTo:
- .
- hosts:
- rds.aws.com
location: MESH_EXTERNAL
ports:
- number: xxxx
   name: tcp
   protocol: TCP
resolution: DNS
